mod-source-record-manager-server/src/main/java/org/folio/verticle/consumers/util/EventTypeHandlerSelector.java

```diff
@@ -0,0 +1,36 @@
+package org.folio.verticle.consumers.util;
+
+import org.folio.DataImportEventPayload;
+import org.folio.DataImportEventTypes;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.folio.DataImportEventTypes.DI_ERROR;
+import static org.folio.DataImportEventTypes.DI_INVOICE_CREATED;
+
+public class EventTypeHandlerSelector {
+
+  public static final String DEFAULT_HANDLER_KEY = "DEFAULT";
+  private final Map<String, SpecificEventHandler> handlers;
+
+  public EventTypeHandlerSelector() {
+    this.handlers = Map.of(DI_INVOICE_CREATED.toString(), new InvoiceImportEventHandler(),
+      DEFAULT_HANDLER_KEY, new MarcImportEventsHandler());
+  }
+
+  public SpecificEventHandler getHandler(DataImportEventPayload eventPayload) {
+    DataImportEventTypes eventType = DataImportEventTypes.valueOf(eventPayload.getEventType());
+    if (DI_INVOICE_CREATED == eventType ||
+      (DI_ERROR == eventType && isLastOperationIs(DI_INVOICE_CREATED.value(), eventPayload.getEventsChain()))) {
```

Comment:
```suggestion
      || ((DI_COMPLETED == eventType ||  DI_ERROR == eventType) 
      && isLastOperationIs(DI_INVOICE_CREATED.value(), eventPayload.getEventsChain()))) {
```
to handle appropriately when import is finished successully

to handle the case when import was successfully completed, in that case, `eventType == DI_COMPLETED` and `DI_INVOICE_CREATED` event is at the end of eventsChain

Reply:
Fixed.